src/confint.jl

```diff
@@ -118,6 +118,10 @@ function confint(bs::BootstrapSample, cim::BCaConfInt, i::Int)
     l = level(cim)
     t0 = original(bs, i)
     t1 = straps(bs, i)
+
+    # Early exit if each element of t1 == t0
+    all(==(t0), t1) && return (t0, t0, t0)
```

Comment:
We only need to check the values of `t1` here, `t0` is technically not relevant:
`var(t1) == 0.0` might be the most idiomatic check here.

Reply:
Should we explictly convert the lower and upper bound to `FloatingPoint`s for type stability?

Reply:
> Should we explictly convert the lower and upper bound to `FloatingPoint`s for type stability?

I don't think that's necessary.

1. I don't imagine folks will be calling this function very often
2. It doesn't seem to make a big difference to performance

Normal run w/o convert:

```
julia> @benchmark confint($bs, $ci)
BenchmarkTools.Trial:
  memory estimate:  58.70 KiB
  allocs estimate:  159
  --------------
  minimum time:     25.226 μs (0.00% GC)
  median time:      30.479 μs (0.00% GC)
  mean time:        45.634 μs (31.37% GC)
  maximum time:     58.377 ms (99.81% GC)
  --------------
  samples:          10000
  evals/sample:     1
```
Normal run w/ convert:
```
julia> @benchmark confint($bs, $ci)
BenchmarkTools.Trial:
  memory estimate:  58.70 KiB
  allocs estimate:  159
  --------------
  minimum time:     21.440 μs (0.00% GC)
  median time:      24.724 μs (0.00% GC)
  mean time:        36.182 μs (27.51% GC)
  maximum time:     46.132 ms (99.81% GC)
  --------------
  samples:          10000
  evals/sample:     1
```

Early exist w/o convert:
```
julia> @benchmark confint($bs, $ci)
BenchmarkTools.Trial:
  memory estimate:  320 bytes
  allocs estimate:  8
  --------------
  minimum time:     1.040 μs (0.00% GC)
  median time:      1.098 μs (0.00% GC)
  mean time:        1.306 μs (9.60% GC)
  maximum time:     1.257 ms (99.77% GC)
  --------------
  samples:          10000
  evals/sample:     10
```
Early exist w/ convert:
```
julia> @benchmark confint($bs, $ci)
BenchmarkTools.Trial:
  memory estimate:  336 bytes
  allocs estimate:  9
  --------------
  minimum time:     1.012 μs (0.00% GC)
  median time:      1.079 μs (0.00% GC)
  mean time:        1.245 μs (6.69% GC)
  maximum time:     833.768 μs (99.81% GC)
  --------------
  samples:          10000
  evals/sample:     12
```
NOTE: I tried this with a few different dataset sizes and also didn't see much of a difference.